users/migrations/0005_transfer_to_pysocial.py

```diff
@@ -0,0 +1,58 @@
+from django.db import connection, migrations
+from django.db.utils import ProgrammingError
+
+
+def transfer_users_from_allauth_to_pysocial(apps, schema_editor):
+    """
+    Transfer user UIDs from Allauth to Python Social Auth.
+    """
+    all_tables = connection.introspection.table_names()
+    if "socialaccount_socialaccount" in all_tables:
+        # allauth
+        with connection.cursor() as cursor:
+            cursor.execute(
+                "SELECT user_id, uid, extra_data FROM socialaccount_socialaccount"
+            )
+            social_accounts = cursor.fetchall()
+
+        # pysocial
+        UserSocialAuth = apps.get_model("social_django", "UserSocialAuth")
+
+        social_auths = []
+        for user_id, uid, extra_data in social_accounts:
```

Comment:
You could just iterate the cursor here to avoid pulling all the users from the DB to the Python memory at once.